src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEReaderTests.cs

```diff
@@ -849,5 +851,45 @@ public void KeepMetadataAlive()
 
             Assert.Equal(@"Debug", reader.GetString(reader.GetAssemblyDefinition().Name));
         }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void InvokeCtorWithIsLoadedImageAndPrefetchMetadataOptions()
+        {
+            using (Process currentProc = Process.GetCurrentProcess())
+            {
+                string libName = "System.Reflection.Metadata.Tests.dll";
+                ProcessModule assemblyModule = currentProc.Modules.OfType<ProcessModule>().First(m => string.Equals(m.ModuleName, libName, StringComparison.Ordinal));
+
+                ulong baseAddress = (ulong)assemblyModule.BaseAddress.ToInt64();
+                int length = assemblyModule.ModuleMemorySize;
+
+                // Read assembly contents out of memory, skipping unmapped pages
+                byte[] assemblyBytes = new byte[length];
+                byte[] page = new byte[PageSize];
+                int bytesRead = 0;
+                for (uint i = 0; i < length; i += PageSize)
+                {
+                    ReadProcessMemory(currentProc.Handle, baseAddress + i, page, PageSize, ref bytesRead);
+                    Array.Copy(page, 0, assemblyBytes, i, bytesRead);
+                }
+
+                using (MemoryStream assemblyStream = new MemoryStream(assemblyBytes))
+                {
+                    PEReader prefetchMetadataReader = new PEReader(assemblyStream, PEStreamOptions.IsLoadedImage | PEStreamOptions.PrefetchMetadata | PEStreamOptions.LeaveOpen);
+                    prefetchMetadataReader.Dispose();
+                }
+            }
+        }
+
+        private const int PageSize = 4096;
+
+        [DllImport("kernel32.dll")]
+        public static extern bool ReadProcessMemory(
+            IntPtr hProcess,
+            ulong lpBaseAddress,
+            byte[] lpBuffer,
+            int nSize,
+            ref int lpNumberOfBytesRead);
```

Comment:
Let's follow the [testing approach used in this library](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.Metadata/tests/TestUtilities/LoaderUtilities.cs)
```suggestion
        [Fact]
        [PlatformSpecific(TestPlatforms.Windows)]  // Uses P/Invokes only suported on windows
        public unsafe void InvokeCtorWithIsLoadedImageAndPrefetchMetadataOptions2()
        {
            using (var tempFile = new TempFile(Path.GetTempFileName()))
            {
                File.WriteAllBytes(tempFile.Path, Misc.Members);

                using (SafeLibraryHandle libHandle = global::Interop.Kernel32.LoadLibraryExW(tempFile.Path, IntPtr.Zero, 0))
                {
                    byte* peImagePtr = (byte*)global::Interop.Kernel32.GetModuleHandle(Path.GetFileName(tempFile.Path));

                    Assert.True(peImagePtr != null);

                    var peReader = new PEReader(new ReadOnlyUnmanagedMemoryStream(peImagePtr, int.MaxValue), PEStreamOptions.IsLoadedImage | PEStreamOptions.PrefetchMetadata);
                    peReader.Dispose();
                }
            }
        }
```